src/org/zaproxy/zap/extension/ascanrulesBeta/BufferOverflow.java

```diff
@@ -0,0 +1,180 @@
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpResponseHeader;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.zaproxy.zap.model.Vulnerabilities;
+import org.zaproxy.zap.model.Vulnerability;
+
+
+public class BufferOverflow extends AbstractAppParamPlugin  {
+
+	/**
+	 * Prefix for internationalised messages used by this rule
+	 */
+	private static final String MESSAGE_PREFIX = "ascanbeta.bufferoverflow.";
+	private static final int PLUGIN_ID = 30001;
+	private static Logger log = Logger.getLogger(BufferOverflow.class);
+	
+	@Override
+	public int getId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return null;
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+	
+
+	@Override
+	public void init() {
+
+	}
+
+	/*
+	 * This method is called by the active scanner for each GET and POST parameter for every page 
+	 * @see org.parosproxy.paros.core.scanner.AbstractAppParamPlugin#scan(org.parosproxy.paros.network.HttpMessage, java.lang.String, java.lang.String)
+	 */
+	@Override
+	public void scan(HttpMessage msg, String param, String value) {
+	
+		if (this.isStop()) { // Check if the user stopped things
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+this.getName()+" Stopping.");
+			}
+			return; // Stop!
+		}
+		
+		try {
+			// This is where you change the 'good' request to attack the application
+			// You can make multiple requests if needed
+			String checkStringHeader1 = "Connection: close";  // Un natural close
+			String errorBufferOverflowMessage = "Potential Buffer Overflow.  The script closed the connection and threw a 500 Internal Server Error";
+			// Always use getNewMsg() for each new request
+			msg = getNewMsg();
+			String returnAttack = randomCharacterString(2100);
+			setParameter(msg, param, returnAttack);
+			sendAndReceive(msg);
+			HttpResponseHeader requestReturn = msg.getResponseHeader();			
+			// This is where BASE baseResponseBody was you detect potential vulnerabilities in the response
+    		String chkerrorheader = requestReturn.getHeadersAsString();
+    		log.debug("Header: "+ chkerrorheader);
+    		if (msg.getResponseHeader().getStatusCode() == HttpStatusCode.INTERNAL_SERVER_ERROR && chkerrorheader.contains(checkStringHeader1))
+    		{
+    			log.debug("Found Header");
+    			bingo(getRisk(), 
+    					Alert.CONFIDENCE_MEDIUM, 
+    					null, 

```

Comment:
It's probably better to use the URI of the request, unless there's a reason not to do so?